Parsers/ASimAuthentication/Parsers/ASimAuthenticationCiscoMeraki.yaml

```diff
@@ -0,0 +1,103 @@
+Parser:
+  Title: ASIM Authentication parser for Cisco Meraki
+  Version: '0.1'
+  LastUpdated: Jun 12 2023
+Product:
+  Name: Cisco Meraki
+Normalization:
+  Schema: Authentication
+  Version: '0.1.3'
+References:
+- Title: ASIM Authentication Schema
+  Link: https://aka.ms/ASimAuthenticationDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing the Cisco Meraki logs to the ASIM authentication normalized schema.
+ParserName: ASimAuthenticationCiscoMeraki
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+
+ParserQuery: |
+  let EventResultDetailsLookup = datatable (reason:string, EventResultDetails:string)
+  [
+      "0","Other",
+      "1","Other",
+      "2","Password expired",
+      "3","Other",
+      "4","Session expired",
+      "5","Other",
+      "6","Other",
+      "7","Other",
+      "8","Other",
+      "9","Other",
+      "10","Logon violates policy",
+      "11","Logon violates policy",
+      "12","Other",
+      "13","Logon violates policy",
+      "14","Other",
+      "15","Other",
+      "16","Other",
+      "17","Other",
+      "18","Incorrect key",
+      "19","Incorrect key",
+      "20","Incorrect key",
+      "21","Other",
+      "22","Other",
+      "23","Other",
+      "24","Logon violates policy",
+  ];
+  let parser = (disabled:bool=false){
+    union isfuzzy=true
+          (
+        meraki_CL
+        | project-rename LogMessage =  Message
+          ),
+          (
+        Syslog
+        | project-rename LogMessage =  SyslogMessage
+          )
+      | where not(disabled) and LogMessage has "events" and (LogMessage has_any ("8021x_auth", "wpa_auth", "splash_auth", "8021x_deauth", "8021x_client_deauth", "wpa_deauth", "8021x_eap_failure", "8021x_eap_success") or (LogMessage has "disassociation" and LogMessage has "auth_neg_failed"))
```

Comment:
I see the logs a following a pattern here.
Timestamp Hostname Type SubType RestOfMessage
Please extract Type and SubType and then perform the exact matching comparison to avoid any discrepancy.

Reply:
In the parser first it identify events from regex we did earlier and then we have added the extraction as you mentioned above to complete match of events.